From 88c8dd8d9e894bcd8343e5ae3b2c8baca4035fa5 Mon Sep 17 00:00:00 2001
From: Ferass El Hafidi <funderscore@postmarketos.org>
Date: Wed, 20 May 2026 16:47:06 +0000
Subject: [PATCH] script: check if console_viewer is enabled

Backport the upstream fix for a NULL console viewer in the script plugin.
The Nabu crash signature is ply_console_viewer_hide() called by
script_lib_sprite_refresh() while the splash is being shown.
---
 src/plugins/splash/script/script-lib-sprite.c | 10 ++++++----
 1 file changed, 6 insertions(+), 4 deletions(-)

diff --git a/src/plugins/splash/script/script-lib-sprite.c b/src/plugins/splash/script/script-lib-sprite.c
index a269ee1..96a01f6 100644
--- a/src/plugins/splash/script/script-lib-sprite.c
+++ b/src/plugins/splash/script/script-lib-sprite.c
@@ -905,10 +905,12 @@ script_lib_sprite_refresh (script_lib_sprite_data_t *data)
                      node;
                      node = ply_list_get_next_node (data->displays, node)) {
                         script_lib_display_t *display = ply_list_node_get_data (node);
-                        if (data->should_show_console_messages == true) {
-                                ply_console_viewer_show (display->console_viewer, display->pixel_display);
-                        } else {
-                                ply_console_viewer_hide (display->console_viewer);
+                        if (display->console_viewer != NULL) {
+                                if (data->should_show_console_messages == true) {
+                                        ply_console_viewer_show (display->console_viewer, display->pixel_display);
+                                } else {
+                                        ply_console_viewer_hide (display->console_viewer);
+                                }
                         }
                 }
                 data->console_viewer_needs_redraw = false;
-- 
2.53.0
